cmake/onnxruntime_providers.cmake

```diff
@@ -753,6 +756,27 @@ if (onnxruntime_USE_NNAPI_BUILTIN)
   endif()
 endif()
 
+if (onnxruntime_USE_SNPE)
+  add_compile_definitions(USE_SNPE=1)
+
+  file(GLOB_RECURSE
+    onnxruntime_providers_snpe_cc_srcs CONFIGURE_DEPENDS
+    "${ONNXRUNTIME_ROOT}/core/providers/snpe/*.h"
+    "${ONNXRUNTIME_ROOT}/core/providers/snpe/*.cc"
+  )
+
+  source_group(TREE ${ONNXRUNTIME_ROOT}/core FILES ${onnxruntime_providers_snpe_cc_srcs})
+  add_library(onnxruntime_providers_snpe ${onnxruntime_providers_snpe_cc_srcs})
+  onnxruntime_add_include_to_target(onnxruntime_providers_snpe onnxruntime_common onnxruntime_framework onnx onnx_proto protobuf::libprotobuf-lite flatbuffers)
+  target_link_libraries(onnxruntime_providers_snpe PRIVATE SNPE.lib)
+  add_dependencies(onnxruntime_providers_snpe onnx ${onnxruntime_EXTERNAL_DEPENDENCIES})
+  set_target_properties(onnxruntime_providers_snpe PROPERTIES CXX_STANDARD_REQUIRED ON)
+  set_target_properties(onnxruntime_providers_snpe PROPERTIES FOLDER "ONNXRuntime")
+  target_include_directories(onnxruntime_providers_snpe PRIVATE ${ONNXRUNTIME_ROOT} ${SNPE_ROOT}/include/zdl)
+  set_target_properties(onnxruntime_providers_snpe PROPERTIES LINKER_LANGUAGE CXX)
+  install(DIRECTORY ${PROJECT_SOURCE_DIR}/../include/onnxruntime/core/providers/snpe  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/onnxruntime/core/providers)
```

Comment:
if the factory header is not needed for snpe, then this line isn't needed right?

Reply:
user need to call OrtSessionOptionsAppendExecutionProvider_SNPE to create Snpe EP, I don't understand why it's not required?

Reply:
see pranav's comment above. all new EP factory functions go directly into c api header.